### PR TITLE
Fix for issue #271 : prevent shorting of already shorten urls

### PR DESCRIPTION
**Pull request by Riyadutt**
issue #271 : https://github.com/origranot/reduced.to/issues/271

Fix: Checking if the host in URL is the same as the host in the request.

### CODE REVIEW

File: backend/src/shortener/shortener.controller.ts [hunk 1]
```diff
@@ -35,6 +36,9 @@ export class ShortenerController {
     }
 
     let shortUrl: string;
+    if(parsedUrl.host === host) {
+      return { newUrl: parsedUrl.href, isExisting: true };
+    }
```
Comment (origranot) [inline backend/src/shortener/shortener.controller.ts:41]: I don't really like this approach of using isExisting variable.
We can just throw a message to the client and show it there.
Also I personally thinks that we shouldn't overload the controller.
Maybe we can try to do a custom decorator with class-validator?

**Review by origranot — CHANGES_REQUESTED** [review]
@Riyadutt 👋 

I left some comments on the pr.
Also, I would like to add a validation on the client side too.

Nice job, I am looking to see the new changes 🚀

File: backend/src/shortener/shortener.controller.ts (hunk 1 above)
Comment (Riyadutt) [reply backend/src/shortener/shortener.controller.ts:41]: sure, will make changes.

**Comment by Riyadutt** [comment]
@origranot - 

> We can just throw a message to the client and show it there.

by this did you mean, to throw an error from API or just send a message key, which can be shown on the front end?

**Comment by Riyadutt** [comment]
> @origranot -
> 
> > We can just throw a message to the client and show it there.
> 
> by this did you mean, to throw an error from API or just send a message key, which can be shown on the front end?

@origranot

**Comment by origranot** [comment]
> @origranot -
> 
> > We can just throw a message to the client and show it there.
> 
> by this did you mean, to throw an error from API or just send a message key, which can be shown on the front end?

@Riyadutt 
I think we should use messages and not error codes.
Let me know if you are still working on the comments 💯

**Comment by origranot** [comment]
Already closed #350

**Pull request closed without merging**